`gcc-4.8 -pedantic` needs explicit `-std=c99` or `-Wno-long-long`

When we give `-pedantic` to `gcc-4.8`, it selects pre-C99.

It causes the following error. ( Here's [actual error log](https://github.com/t-mat/xxHash/runs/3345251136#step:9:23) )

```
gcc-4.8 -O3 -Wall -Wextra -Wconversion -Wcast-qual -Wcast-align -Wshadow -Wstrict-aliasing=1 -Wswitch-enum -Wdeclaration-after-statement -Wstrict-prototypes -Wundef -Wpointer-arith -Wformat-security -Wvla -Wformat=2 -Winit-self -Wfloat-equal -Wwrite-strings -Wredundant-decls -Wstrict-overflow=2  -Werror -pedantic -DXXH_NO_XXH3 xxhash.c -c
In file included from xxhash.c:43:0:
xxhash.h:572:41: error: use of C99 long long integer constant [-Werror=long-long]
 #  if defined(__LP64__) && ULONG_MAX == 0xFFFFFFFFFFFFFFFFULL
                                         ^
```

There're two reasons why we only see this error in `make noxxh3test` with `gcc-4.8`.

(1) Default C version of `gcc-4.8 -pedantic` is pre-C99.

```
gcc 4.8.5 -pedantic     __STDC_VERSION__ = undefined  (technically, pre-C89?)
gcc 5.1.0 -pedantic     __STDC_VERSION__ = 201112
gcc 5.2.0 -pedantic     __STDC_VERSION__ = 201112
...
```

(2) `make noxxh3test` doesn't give `-std`.

`make c90test` gives `-pedantic -std=c90` (edit: and `-DXXH_NO_LONG_LONG` ).  Therefore it has no error.

```
c90test: CPPFLAGS += -DXXH_NO_LONG_LONG
c90test: CFLAGS += -std=c90 -Werror -pedantic
```

But `make noxxh3test` only gives `-pedantic`.  It causes this problem.

```
noxxh3test: CFLAGS += -Werror -pedantic
```

----

I think adding `-std=c99` to `noxxh3test` is the most simple way to fix it.  But I'm not sure it breaks intention of this test or not.

edit: `c90` -> `c99`

--- a/Makefile
+++ b/Makefile
@@ -318,7 +318,7 @@ c90test: xxhash.c
 endif
 
 noxxh3test: CPPFLAGS += -DXXH_NO_XXH3
-noxxh3test: CFLAGS += -Werror -pedantic
+noxxh3test: CFLAGS += -Werror -pedantic -Wno-long-long  # XXH64 requires long long support
 noxxh3test: xxhash.c
 	@echo ---- test compilation without XXH3 ----
 	$(RM) xxhash.o

